### PR TITLE
Cad: Generalize src/* IFC references to prepare for .obj support.

### PR DESCRIPTION
**Pull request by pablo-mayrgundter**
Plan of attack here is to de-IFC-ize things as much as possible then push down the IfcLoader into either a subclass of CadView, or a hook or something.

I'm leaving "IFC" in comments when there's full path examples and to minimize change, but this could change here too if it's jarring.

### CODE REVIEW

File: src/Components/About/AboutControl.jsx [hunk 1]
```diff
@@ -28,7 +28,6 @@ export default function AboutControl() {
     setIsDialogDisplayed(value)
   }
 
-  // eslint-disable-next-line no-unused-vars
```
Comment (pablo-mayrgundter) [inline src/Components/About/AboutControl.jsx:31]: This showed up in eslint as unused.

File: src/MimeType.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  if (lastDotNdx === -1) {
+    return false
+  }
+  return isExtensionSupported(pathWithSuffix.substring(lastDotNdx + 1))
+}
+
+
+/**
+ * @param {string} filepath
+ * @return {Array.<string>}
```
Comment (oo-bldrs) [inline src/MimeType.js:28]: This function returns an object that consists of two members which are arrays.

File: src/Share.jsx [hunk 3]
```diff
@@ -125,12 +133,7 @@ export function getModelPath(installPrefix, pathPrefix, urlParams) {
   if (filepath === '') {
     return null
   }
-  const splitRegex = /\.ifc/i
-  const match = splitRegex.exec(filepath)
-  if (!match) {
-    throw new Error('Filepath must contain ".ifc" (case-insensitive)')
-  }
-  const parts = filepath.split(splitRegex)
+  const {parts, match} = splitAroundExtension(filepath)
```
Comment (oo-bldrs) [inline src/Share.jsx:136]: `splitAroundExtension` raises an exception if an unsupported file path is passed to it, which is not caught here?

Does it make sense to perform a check with `isExtensionSupported` prior to performing the split, or handling the exception, or adding a `TODO` for the future?

**Review by oo-bldrs — DISMISSED** [review]

File: src/MimeType.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+ */
+export function splitAroundExtension(filepath) {
+  const splitRegex = /\.(?:ifc|obj)/i
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/MimeType.js:31]: Done.

File: src/MimeType.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+  const match = splitRegex.exec(filepath)
+  if (!match) {
+    throw new Error('Filepath must contain ".(ifc|obj)" (case-insensitive)')
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/MimeType.js:34]: Done

File: src/MimeType.js (hunk 2 above)
Comment (pablo-mayrgundter) [reply src/MimeType.js:28]: Done

File: src/MimeType.js
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/MimeType.js:1]: Done.

File: src/MimeType.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
+export const supportedTypes = ['ifc', 'obj']
```
[the remark this answers is not in the document]
Comment (pablo-mayrgundter) [reply src/MimeType.js]: Done.

File: src/Share.jsx (hunk 3 above)
Comment (pablo-mayrgundter) [reply src/Share.jsx]: Good catch.  I restored use of `alert` and bubble it up.

Done.

**Comment by pablo-mayrgundter** [comment]
@oo-bldrs Sorry, lost track of this.  Updated now getting ready for conway loader.   All comments addressed.

**Comment by pablo-mayrgundter** [comment]
btw, cypress runs fine locally for me but is taking 40+ mins here.. 
image

**Review by nickcastel50 — APPROVED** [review]
LGTM!

**Pull request merged**